File: src/test/java/doc/java/examples/filters/MyCommandFilters.java
package doc.java.examples.filters;

import io.github.freya022.botcommands.api.commands.application.ApplicationCommandFilter;
import io.github.freya022.botcommands.api.commands.application.ApplicationCommandInfo;
import io.github.freya022.botcommands.api.commands.text.TextCommandFilter;
import io.github.freya022.botcommands.api.commands.text.TextCommandVariation;
import io.github.freya022.botcommands.api.core.service.annotations.BService;
import io.github.freya022.botcommands.test.switches.TestLanguage;
import io.github.freya022.botcommands.test.switches.TestService;
import net.dv8tion.jda.api.entities.channel.Channel;
import net.dv8tion.jda.api.events.interaction.command.GenericCommandInteractionEvent;
import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@BService
@TestService
@TestLanguage(TestLanguage.Language.JAVA)
public class MyCommandFilters implements TextCommandFilter<String>, ApplicationCommandFilter<String> {
    @Nullable
    @Override
    public String check(@NotNull MessageReceivedEvent event, @NotNull TextCommandVariation commandVariation, @NotNull String args) {
        return check(event.getGuildChannel());
    }

    @Nullable
    @Override
    public String check(@NotNull GenericCommandInteractionEvent event, @NotNull ApplicationCommandInfo commandInfo) {
        return check(event.getGuildChannel());
    }

    @Nullable
    private String check(@NotNull Channel channel) {
        if (channel.getIdLong() != 722891685755093076L) {
            return "Can only run commands in <#722891685755093076>";
        }
        return null;
    }
}

File: src/test/kotlin/io/github/freya022/botcommands/test/filters/MyCommandFilters.kt
package io.github.freya022.botcommands.test.filters

import dev.minn.jda.ktx.coroutines.await
import dev.minn.jda.ktx.messages.reply_
import io.github.freya022.botcommands.api.commands.application.ApplicationCommandFilter
import io.github.freya022.botcommands.api.commands.application.ApplicationCommandRejectionHandler
import io.github.freya022.botcommands.api.commands.text.TextCommandFilter
import io.github.freya022.botcommands.api.commands.text.TextCommandRejectionHandler
import io.github.freya022.botcommands.api.core.service.annotations.BService
import io.github.freya022.botcommands.internal.commands.application.ApplicationCommandInfo
import io.github.freya022.botcommands.internal.commands.prefixed.TextCommandVariation
import net.dv8tion.jda.api.entities.channel.Channel
import net.dv8tion.jda.api.events.interaction.command.GenericCommandInteractionEvent
import net.dv8tion.jda.api.events.message.MessageReceivedEvent

@BService
class MyCommandFilters : TextCommandFilter, ApplicationCommandFilter {
    override suspend fun checkSuspend(
        event: MessageReceivedEvent,
        commandVariation: TextCommandVariation,
        args: String
    ): Any? = check(event.guildChannel)

    override suspend fun checkSuspend(
        event: GenericCommandInteractionEvent,
        commandInfo: ApplicationCommandInfo
    ): Any? = check(event.guildChannel)

    private fun check(channel: Channel): String? {
        if (channel.idLong != 722891685755093076) {
            return "Can only run commands in <#722891685755093076>"
        }
        return null
    }
}

@BService
class MyCommandRejectionHandler : TextCommandRejectionHandler, ApplicationCommandRejectionHandler {
    override suspend fun handleSuspend(
        event: MessageReceivedEvent,
        variation: TextCommandVariation,
        args: String,
        userData: Any
    ) {
        event.message.reply(userData as String).await()
    }

    override suspend fun handleSuspend(
        event: GenericCommandInteractionEvent,
        commandInfo: ApplicationCommandInfo,
        userData: Any
    ) {
        event.reply_(userData as String, ephemeral = true).await()
    }
}